### PR TITLE
g3-2024-v5-#14885

### PR DESCRIPTION
**Pull request by a22samka**

### CODE REVIEW

File: Shared/toast.php [hunk 1]
```diff
@@ -102,6 +102,54 @@ function toast(type, text, duration, arguments = null) {
...
 
+        function truncateOverflow(toastText) {
+        const maxChars = 70; // Set the maximum number of characters before truncating
+        const toastContent = toastText.textContent.trim();
+        if (toastContent.length > maxChars) {
+            // Truncate the content and add "..."
+            toastText.textContent = toastContent.slice(0, maxChars) + '...';
+            toastText.dataset.fullText = toastContent; // Store the full text in a data attribute
+            }
+        }
```
Comment (a22moamy) [inline Shared/toast.php:113]: indentation

File: Shared/toast.php [hunk 2]
```diff
@@ -102,6 +102,54 @@ function toast(type, text, duration, arguments = null) {
...
+
+
+            }
+        });
+
+
+        truncateOverflow(toastText); // Truncate overflow text initially
+
+
+
```
Comment (a22moamy) [inline Shared/toast.php:152]: I won't comment for every unnecessary white space, but you should generally get rid of all of them

File: Shared/toast.php [hunk 3]
```diff
@@ -102,6 +102,54 @@ function toast(type, text, duration, arguments = null) {
...
+                    {
+                        // Reset the height to auto to show full text
+                        toastText.style.height = 'auto';
+                    }
+                      // If the toast was previously expanded and is now collapsed, reset the height to 2rem
+                    if (isExpanded && !this.classList.contains('expanded'))
+                    {
+                        truncateOverflow(toastText); // Truncate overflow text
+                        toastText.style.height = '2.3rem';
+                    }
```
Comment (a22moamy) [inline Shared/toast.php:142]: Have the "{"-symbol be on the same line as the if statement. This way we are following the K&R code standard. [For more info](https://en.wikipedia.org/wiki/Indentation_style).

Comment (a22moamy) [inline Shared/toast.php:142]: Indentation

File: Shared/toast.php [hunk 4]
```diff
@@ -102,6 +102,54 @@ function toast(type, text, duration, arguments = null) {
...
+            toastText.dataset.fullText = toastContent; // Store the full text in a data attribute
+            }
+        }
+
+
+
+
+        toastDiv.addEventListener('click', function(event) {
+            // Check if the click target is not the close button
+            if (!event.target.classList.contains('closeIcon')) {
```
Comment (a22moamy) [inline Shared/toast.php:120]: indentation

File: Shared/toast.php [hunk 5]
```diff
@@ -102,6 +102,54 @@ function toast(type, text, duration, arguments = null) {
...
+                toastText.textContent = toastText.dataset.fullText || toastText.textContent;
+                   
+                    // If the toast is expanded, set the height to auto to show full text
+                    if (this.classList.contains('expanded'))
+                    {
+                        // Reset the height to auto to show full text
+                        toastText.style.height = 'auto';
+                    }
+                      // If the toast was previously expanded and is now collapsed, reset the height to 2rem
+                    if (isExpanded && !this.classList.contains('expanded'))
```
Comment (a22moamy) [inline Shared/toast.php:138]: indentation

File: Shared/toast.php [hunk 6]
```diff
@@ -102,6 +102,54 @@ function toast(type, text, duration, arguments = null) {
...
+
+                this.classList.toggle('expanded');
+                toastText.textContent = toastText.dataset.fullText || toastText.textContent;
+                   
+                    // If the toast is expanded, set the height to auto to show full text
+                    if (this.classList.contains('expanded'))
+                    {
+                        // Reset the height to auto to show full text
+                        toastText.style.height = 'auto';
+                    }
```
Comment (a22moamy) [inline Shared/toast.php:136]: indentation

**Review by a22moamy — CHANGES_REQUESTED** [review]
I have not checked the functionality of your pull request, but you have gotten some change requests regarding indentation and white spaces

**Comment by a22moamy** [comment]
Please add a description of your pull request next time as well, along with information regarding how to test it (for future testers).

File: Shared/toast.php [hunk 7]
```diff
@@ -102,6 +102,40 @@ function toast(type, text, duration, arguments = null) {
...
+                toastText.style.height = 'auto';
+            }
+            // If the toast was previously expanded and is now collapsed, reset the height to 2rem
+            if (isExpanded && !this.classList.contains('expanded')){
+                truncateOverflow(toastText); // Truncate overflow text
+                toastText.style.height = '2.3rem';
+            }
+        }
+    });
+    truncateOverflow(toastText); // Truncate overflow text initially
```
Comment (a22moamy) [inline Shared/toast.php:137]: Indent once. This code is on the same indentation level as the whole toast function for me

File: Shared/toast.php [hunk 8]
```diff
@@ -102,6 +102,40 @@ function toast(type, text, duration, arguments = null) {
...
 
+        function truncateOverflow(toastText) {
+            const maxChars = 70; // Set the maximum number of characters before truncating
+            const toastContent = toastText.textContent.trim();
+            if (toastContent.length > maxChars) {
+                // Truncate the content and add "..."
+                toastText.textContent = toastContent.slice(0, maxChars) + '...';
+                toastText.dataset.fullText = toastContent; // Store the full text in a data attribute
+            }
+        }
```
Comment (a22moamy) [inline Shared/toast.php:113]: Do not define and declare a function inside another function. Instead, move the truncateOverflow()-function outside the toast()-function.

File: Shared/css/style.css [hunk 9]
```diff
@@ -9653,16 +9653,22 @@ margin:2px;
...
-  height: 3rem;
+  min-height:2.3rem;
+  height: 2.3rem;
   word-break: break-word;
-  text-align: start; 
+  text-align: start;
+  overflow: hidden;
 }
 
+
```
Comment (a22moamy) [inline Shared/css/style.css:9665]: In the future, avoid having two whitespaces after another. Just one will do. For now you don't need to change this, just letting you know. Makes the code easier to read through for future students.

**Review by a22moamy — CHANGES_REQUESTED** [review]
See comments

**Review by a22moamy — APPROVED** [review]
Code looks good now, haven't tested the functionality though. 😄

**Review by a22oscja — APPROVED** [review]
tested functionality, works as it should.

**Pull request merged**